pkg/daemon/update.go

```diff
@@ -574,27 +574,33 @@ func checkFIPS(current, desired *mcfgv1.MachineConfig) error {
 	return errors.New("detected change to FIPS flag. Refusing to modify FIPS on a running cluster")
 }
 
+// parseKernelArguments separates out kargs from each entry and returns it as a map for
+// easy comparision
+func parseKernelArguments(kargs []string) map[string]bool {
+	parsed := make(map[string]bool)
+	for _, k := range kargs {
+		for _, arg := range strings.Split(k, " ") {
```

Comment:
Isn't this too simplistic?  Is this going to work for quoted kernel parameters with spaces in them such as `param1="hello world!" param2=helllo world-param3`?  I'd suggest looking at kernel code what they do, in particular `next_arg()` function.  I believe something like this might do a better job:
```
// checks for white-space characters in "C" and "POSIX" locales.
func isspace(b byte) bool {
	return b == ' ' || b == '\f' || b == '\n' || b == '\r' || b == '\t' || b == '\v'
}

// You can use " around spaces, but can't escape ".  See next_arg() in kernel code.
func nextArg(args string, start int) (int, int) {
	var (
		i, equals int
		stop int
		inQuote bool
	)

	// Skip leading spaces
	for start < len(args) && isspace(args[start]) {
		start++
	}

	i = start

	if (start < len(args) && args[i] == '"') {
		i++
		inQuote = true
	}

	for ; i < len(args); i++ {
		if isspace(args[i]) && !inQuote {
			break
		}

		if equals == 0 && args[i] == '=' {
			equals = i
		}

		if args[i] == '"' {
			inQuote = !inQuote
		}
	}

	stop = i

	return start, stop
}

// SplitKernelArguments splits kernel parameters into a slice of strings one parameter per string.
func SplitKernelArguments(args string) ([]string) {
	var (
		start, stop int
		kv []string
	)

	for stop < len(args) {
		start, stop = nextArg(args, stop)
		if start != stop {
			kv = append(kv, args[start:stop])
		}
	}

	return kv
}
```

Reply:
Yes, this is the naive first attempt. I'm looking into copying over the next_arg logic now.

Reply:
Updated with your suggestion. Confirmed the logic matches the kernel code 👍